### PR TITLE
Add back the timeout for TelemetryClient.Flush to avoid deadlock

### PR DESCRIPTION
**Pull request by 928PJY**
#512461

### CODE REVIEW

File: src/docfx/lib/log/Telemetry.cs [hunk 1]
```diff
@@ -233,7 +234,8 @@ public static void TrackException(Exception ex)
 
         public static void Flush()
         {
-            s_telemetryClient.Flush();
+            // TelemetryClient.Flush may meet deadlocks: https://github.com/microsoft/ApplicationInsights-dotnet/issues/1186
+            Task.WaitAny(Task.Run(s_telemetryClient.Flush), Task.Delay(2000));
```
Comment (yufeih) [inline src/docfx/lib/log/Telemetry.cs:238]: We can increase the timeout a little bit if 2 seconds is to small to flush all the telemetry.

**Review by yufeih — APPROVED** [review]

File: src/docfx/lib/log/Telemetry.cs (hunk 1 above)
Comment (928PJY) [reply src/docfx/lib/log/Telemetry.cs:238]: like 10 seconds?

Comment (yufeih) [reply src/docfx/lib/log/Telemetry.cs:238]: Sounds good

**Review by superyyrrzz — APPROVED** [review]

**Comment by superyyrrzz** [comment]
Does this change need to be reverted? https://github.com/928PJY/docfx/commit/de7590b1e6b647243045f1f57fdd18f9cad145fb#diff-3dc856ced277d6fc688a197540a9b355eca4dca734689ae41a18b342523902dd

**Pull request merged**